### ISSUE TITLE
Icons not rendering in RStudio Viewer

### ISSUE THREAD

**Issue by jmgirard** [jmgirard/lordicon#1]
Icons are rendering with Firefox and Chrome but not the RStudio Viewer.

**Comment by jmgirard** [comment]
RStudio Viewer error message: 
> Uncaught SyntaxError: Unexpected token ;

**Comment by jmgirard** [comment]
When bundled on development mode, the first error is attributed to this line:
```
"use strict";
 eval("__webpack_require__.r(__webpack_exports__);\n/* harmony export */ __webpack_require__.d(__webpack_exports__, {\n/* harmony export */   \"Element\": () => (/* binding */ Element)\n/* harmony export */ });\n/* harmony import */ var _helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__ = __webpack_require__(/*! ../helpers/lottie.js */ \"./node_modules/lord-icon-element/build/helpers/lottie.js\");\n/* harmony import */ var _helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__ = __webpack_require__(/*! ../helpers/utils.js */ \"./node_modules/lord-icon-element/build/helpers/utils.js\");\n/* harmony import */ var _manager_js__WEBPACK_IMPORTED_MODULE_0__ = __webpack_require__(/*! ./manager.js */ \"./node_modules/lord-icon-element/build/main/manager.js\");\n/* harmony import */ var _global_js__WEBPACK_IMPORTED_MODULE_1__ = __webpack_require__(/*! ../global.js */ \"./node_modules/lord-icon-element/build/global.js\");\n\n\n\n\n/**\n * Loads lottie dom elements when needed.\n */\nconst PROGRESSIVE_LOAD = false;\n/**\n * Prefix for icon states.\n */\nconst STATE_PREFIX = 'State-';\n/**\n * Use constructable stylesheets if supported (https://developers.google.com/web/updates/2019/02/constructable-stylesheets)\n */\nconst SUPPORTS_ADOPTING_STYLE_SHEETS = \"adoptedStyleSheets\" in Document.prototype && \"replace\" in CSSStyleSheet.prototype;\n/**\n * Style for this element.\n */\nconst ELEMENT_STYLE = `\n    :host {\n      display: inline-flex;\n      width: 32px;\n      height: 32px;\n      align-items: center;\n      justify-content: center;\n      position: relative;\n      vertical-align: middle;\n      overflow: hidden;\n    }\n\n    :host(.current-color) svg path[fill] {\n      fill: currentColor;\n    }\n\n    :host(.current-color) svg path[stroke] {\n      stroke: currentColor;\n    }\n\n    :host(:not(.current-color)) svg .primary path[fill] {\n      fill: var(--lord-icon-primary, var(--lord-icon-primary-base));\n    }\n\n    :host(:not(.current-color)) svg .primary path[stroke] {\n      stroke: var(--lord-icon-primary, var(--lord-icon-primary-base));\n    }\n\n    :host(:not(.current-color)) svg .secondary path[fill] {\n      fill: var(--lord-icon-secondary, var(--lord-icon-secondary-base));\n    }\n\n    :host(:not(.current-color)) svg .secondary path[stroke] {\n      stroke: var(--lord-icon-secondary, var(--lord-icon-secondary-base));\n    }\n\n    svg {\n      pointer-events: none;\n      display: block;\n    }\n\n    div {\n      width: 100%;\n      height: 100%;\n    }\n\n    div.slot {\n      position: absolute;\n      left: 0;\n      top: 0;\n      z-index: 2;\n    }\n`;\n/**\n * Current style sheet.\n */\nlet styleSheet;\n/**\n * Observed attributes for this custom element.\n */\nconst OBSERVED_ATTRIBUTES = [\n    \"colors\",\n    \"src\",\n    \"icon\",\n    \"state\",\n    \"trigger\",\n    \"speed\",\n    \"stroke\",\n    \"scale\",\n    \"axis-x\",\n    \"axis-y\",\n];\nclass Element extends HTMLElement {\n    _root;\n    _isReady = false;\n    _lottie;\n    _properties;\n    _connectedTrigger;\n    _storedIconData;\n    _palette;\n    /**\n     * Register Lottie library.\n     * @param animationLoader Provide \"loadAnimation\" here from Lottie.\n     */\n    static registerAnimationLoader(animationLoader) {\n        (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.registerAnimationLoader)(animationLoader);\n    }\n    /**\n     * Register supported icon. This is helpful with any kind of preload icons.\n     * @param name Icon name.\n     * @param iconData Icon data.\n     */\n    static registerIcon(name, iconData) {\n        (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.registerIcon)(name, iconData);\n    }\n    /**\n     * Register supported animation.\n     * @param name\n     * @param triggerClass\n     */\n    static registerTrigger(name, triggerClass) {\n        (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.registerTrigger)(name, triggerClass);\n    }\n    /**\n     * Custom element observed attributes.\n     */\n    static get observedAttributes() {\n        return OBSERVED_ATTRIBUTES;\n    }\n    /**\n     * Check element version.\n     */\n    static get version() {\n        return _global_js__WEBPACK_IMPORTED_MODULE_1__.VERSION;\n    }\n    constructor() {\n        super();\n        // create shadow root for this element\n        this._root = this.attachShadow({\n            mode: \"open\"\n        });\n    }\n    /**\n     * Element connected.\n     */\n    connectedCallback() {\n        (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.connectInstance)(this);\n        // execute init only once after connected\n        if (!this._isReady) {\n            this.init();\n        }\n    }\n    /**\n     * Element disconnected.\n     */\n    disconnectedCallback() {\n        this.unregisterLottie();\n        (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.disconnectInstance)(this);\n    }\n    /**\n     * Handle attribute update.\n     * @param name\n     * @param oldValue\n     * @param newValue\n     */\n    attributeChangedCallback(name, oldValue, newValue) {\n        if (name === 'axis-x') {\n            this.axisXChanged();\n        }\n        else if (name === 'axis-y') {\n            this.axisYChanged();\n        }\n        else {\n            const method = this[`${name}Changed`];\n            if (method) {\n                method.call(this);\n            }\n        }\n    }\n    /**\n     * Init element.\n     * @returns\n     */\n    init() {\n        if (this._isReady) {\n            return;\n        }\n        this._isReady = true;\n        if (SUPPORTS_ADOPTING_STYLE_SHEETS) {\n            if (!styleSheet) {\n                styleSheet = new CSSStyleSheet();\n                styleSheet.replaceSync(ELEMENT_STYLE);\n            }\n            this._root.adoptedStyleSheets = [styleSheet];\n        }\n        else {\n            const style = document.createElement(\"style\");\n            style.innerHTML = ELEMENT_STYLE;\n            this._root.appendChild(style);\n        }\n        const slotContainer = document.createElement(\"div\");\n        slotContainer.innerHTML = \"<slot></slot>\";\n        slotContainer.classList.add(\"slot\");\n        this._root.appendChild(slotContainer);\n        const container = document.createElement(\"div\");\n        container.classList.add('body');\n        this._root.appendChild(container);\n        this.registerLottie();\n    }\n    registerLottie() {\n        let iconData = this.iconData;\n        if (!iconData) {\n            return;\n        }\n        this._lottie = (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.loadLottieAnimation)({\n            container: this.container,\n            renderer: \"svg\",\n            loop: false,\n            autoplay: false,\n            animationData: (0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.deepClone)(iconData),\n            rendererSettings: {\n                preserveAspectRatio: \"xMidYMid meet\",\n                progressiveLoad: PROGRESSIVE_LOAD,\n                hideOnTransparent: true,\n            },\n        });\n        if (this.state || this.colors || this.stroke || this.scale || this.axisX || this.axisY) {\n            const properties = this.properties;\n            if (properties) {\n                if (this.colors) {\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateColors)(this._lottie, properties, this.colors);\n                }\n                if (this.state) {\n                    for (const state of this.states) {\n                        (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.replaceProperty)(this._lottie, properties, STATE_PREFIX + state, 0);\n                    }\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.replaceProperty)(this._lottie, properties, STATE_PREFIX + this.state, 1);\n                }\n                if (this.stroke) {\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, properties, 'stroke', this.stroke);\n                }\n                if (this.scale) {\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, properties, 'scale', this.scale);\n                }\n                if (this.axisX) {\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, properties, 'axis', this.axisX, '0');\n                }\n                if (this.axisY) {\n                    (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, properties, 'axis', this.axisY, '1');\n                }\n                this._lottie.renderer.renderFrame(null);\n            }\n        }\n        // set speed\n        this._lottie.setSpeed(this.animationSpeed);\n        // dispatch animation-complete\n        this._lottie.addEventListener(\"complete\", () => {\n            this.dispatchEvent(new CustomEvent(\"animation-complete\"));\n        });\n        this.triggerChanged();\n        this.dispatchEvent(new CustomEvent(\"icon-ready\"));\n        // move palette to css variables instantly on this icon\n        if ((0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.iconFeatures)(iconData).includes('css-variables')) {\n            this.movePaletteToCssVariables();\n        }\n    }\n    unregisterLottie() {\n        this._properties = undefined;\n        if (this._connectedTrigger) {\n            this._connectedTrigger.disconnectedCallback();\n            this._connectedTrigger = undefined;\n        }\n        if (this._lottie) {\n            this._lottie.destroy();\n            this._lottie = undefined;\n            this.container.innerHTML = \"\";\n        }\n    }\n    refresh() {\n        this._lottie?.renderer.renderFrame(null);\n        this.movePaletteToCssVariables();\n    }\n    notify(name, from) {\n        if (this[from] !== name) {\n            return;\n        }\n        if (from === \"icon\") {\n            if (this._lottie) {\n                this.unregisterLottie();\n            }\n            this.registerLottie();\n        }\n        else if (from === \"trigger\" && !this._connectedTrigger) {\n            this.triggerChanged();\n        }\n    }\n    triggerChanged() {\n        if (this._connectedTrigger) {\n            this._connectedTrigger.disconnectedCallback();\n            this._connectedTrigger = undefined;\n        }\n        if (this.trigger && this._lottie) {\n            const TriggerClass = (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.getTrigger)(this.trigger);\n            if (TriggerClass) {\n                this._connectedTrigger = new TriggerClass(this, this._lottie);\n                this._connectedTrigger.connectedCallback();\n            }\n        }\n    }\n    colorsChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if (this.colors) {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateColors)(this._lottie, this.properties, this.colors);\n        }\n        else {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetColors)(this._lottie, this.properties);\n        }\n        this.refresh();\n    }\n    strokeChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(this.stroke)) {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetProperty)(this._lottie, this.properties, 'stroke');\n        }\n        else {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, this.properties, 'stroke', this.stroke);\n        }\n        this.refresh();\n    }\n    stateChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if (this.state) {\n            for (const state of this.states) {\n                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.replaceProperty)(this._lottie, this.properties, STATE_PREFIX + state, 0);\n            }\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.replaceProperty)(this._lottie, this.properties, STATE_PREFIX + this.state, 1);\n        }\n        else {\n            for (const state of this.states) {\n                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetProperty)(this._lottie, this.properties, STATE_PREFIX + state);\n            }\n        }\n        this.refresh();\n    }\n    scaleChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(this.scale)) {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetProperty)(this._lottie, this.properties, 'scale');\n        }\n        else {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, this.properties, 'scale', this.scale);\n        }\n        this.refresh();\n    }\n    axisXChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(this.axisX)) {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetProperty)(this._lottie, this.properties, 'axis', '0');\n        }\n        else {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, this.properties, 'axis', this.axisX, '0');\n        }\n        this.refresh();\n    }\n    axisYChanged() {\n        if (!this._isReady || !this.properties) {\n            return;\n        }\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(this.axisY)) {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetProperty)(this._lottie, this.properties, 'axis', '1');\n        }\n        else {\n            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateProperty)(this._lottie, this.properties, 'axis', this.axisY, '1');\n        }\n        this.refresh();\n    }\n    speedChanged() {\n        if (this._lottie) {\n            this._lottie.setSpeed(this.animationSpeed);\n        }\n    }\n    iconChanged() {\n        if (!this._isReady) {\n            return;\n        }\n        this.unregisterLottie();\n        this.registerLottie();\n    }\n    async srcChanged() {\n        if (this.src) {\n            await (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.loadIcon)(this.src);\n        }\n        if (!this._isReady) {\n            return;\n        }\n        this.unregisterLottie();\n        this.registerLottie();\n    }\n    movePaletteToCssVariables() {\n        for (const [key, value] of Object.entries(this.palette)) {\n            this._root.querySelector('.body').style.setProperty(`--lord-icon-${key}-base`, value);\n        }\n    }\n    /**\n     * Access current trigger instance.\n     */\n    get connectedTrigger() {\n        return this._connectedTrigger;\n    }\n    /**\n     * Available properties for current icon.\n     */\n    get properties() {\n        if (!this._properties && this.iconData) {\n            this._properties = (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.allProperties)(this.iconData, true);\n        }\n        return this._properties || [];\n    }\n    /**\n     * Available states for current icon.\n     */\n    get states() {\n        return this.properties.filter(c => c.name.startsWith(STATE_PREFIX)).map(c => {\n            return c.name.substr(STATE_PREFIX.length).toLowerCase();\n        });\n    }\n    /**\n     * Find default state.\n     */\n    get defaultState() {\n        const states = this.properties.filter(c => c.name.startsWith(STATE_PREFIX) && c.value);\n        if (states.length) {\n            return states[0].name.substr(STATE_PREFIX.length).toLowerCase();\n        }\n        return undefined;\n    }\n    /**\n     * Check whether the element is ready.\n     */\n    get isReady() {\n        return this._isReady;\n    }\n    /**\n     * Access lottie animation instance.\n     */\n    get lottie() {\n        return this._lottie;\n    }\n    /**\n     * Update palette.\n     */\n    set palette(colors) {\n        if (!colors || typeof colors !== 'object') {\n            return;\n        }\n        for (const current of this.properties) {\n            if (current.type !== 'color') {\n                continue;\n            }\n            const name = current.name.toLowerCase();\n            if (name in colors && colors[name]) {\n                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateColor)(this._lottie, this.properties, name, colors[name]);\n            }\n            else {\n                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetColor)(this._lottie, this.properties, name);\n            }\n        }\n        this.refresh();\n    }\n    /**\n     * Access to colors get / update with convenient way.\n     */\n    get palette() {\n        if (!this._palette) {\n            this._palette = new Proxy(this, {\n                set: (target, property, value, receiver) => {\n                    for (const current of target.properties) {\n                        if (current.type == 'color' && typeof property === 'string' && property.toLowerCase() == current.name.toLowerCase()) {\n                            if (value) {\n                                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.updateColor)(target._lottie, target.properties, property, value);\n                            }\n                            else if (value === undefined) {\n                                (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetColor)(target._lottie, target.properties, property);\n                            }\n                            target.refresh();\n                        }\n                    }\n                    return true;\n                },\n                get: (target, property, receiver) => {\n                    for (const current of target.properties) {\n                        if (current.type == 'color' && typeof property === 'string' && property.toLowerCase() == current.name.toLowerCase()) {\n                            return (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.lottieColorToHex)((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.get)(target._lottie, current.path));\n                        }\n                    }\n                    return undefined;\n                },\n                deleteProperty: (target, property) => {\n                    for (const current of target.properties) {\n                        if (current.type == 'color' && typeof property === 'string' && property.toLowerCase() == current.name.toLowerCase()) {\n                            (0,_helpers_lottie_js__WEBPACK_IMPORTED_MODULE_3__.resetColor)(target._lottie, target.properties, property);\n                            target.refresh();\n                        }\n                    }\n                    return true;\n                },\n                ownKeys: (target) => {\n                    return target.properties.filter(c => c.type == 'color').map(c => c.name.toLowerCase());\n                },\n                has: (target, property) => {\n                    for (const current of target.properties) {\n                        if (current.type == 'color' && typeof property === 'string' && property.toLowerCase() == current.name.toLowerCase()) {\n                            return true;\n                        }\n                    }\n                    return false;\n                },\n                getOwnPropertyDescriptor: (target) => {\n                    return {\n                        enumerable: true,\n                        configurable: true,\n                    };\n                },\n            });\n        }\n        return this._palette;\n    }\n    set icon(value) {\n        if (value && (0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isObjectLike)(value)) {\n            this._storedIconData = value;\n            if (this.hasAttribute('icon')) {\n                this.removeAttribute('icon');\n            }\n            else {\n                this.iconChanged();\n            }\n        }\n        else {\n            const oldIconData = this._storedIconData;\n            this._storedIconData = undefined;\n            if (value) {\n                this.setAttribute('icon', value);\n            }\n            else {\n                this.removeAttribute('icon');\n                if (oldIconData) {\n                    this.iconChanged();\n                }\n            }\n        }\n    }\n    get icon() {\n        return this._storedIconData || this.getAttribute('icon');\n    }\n    set src(value) {\n        if (value) {\n            this.setAttribute('src', value);\n        }\n        else {\n            this.removeAttribute('src');\n        }\n    }\n    get src() {\n        return this.getAttribute('src');\n    }\n    set state(value) {\n        if (value) {\n            this.setAttribute('state', value);\n        }\n        else {\n            this.removeAttribute('state');\n        }\n    }\n    get state() {\n        return this.getAttribute('state');\n    }\n    set colors(value) {\n        if (value) {\n            this.setAttribute('colors', value);\n        }\n        else {\n            this.removeAttribute('colors');\n        }\n    }\n    get colors() {\n        return this.getAttribute('colors');\n    }\n    set trigger(value) {\n        if (value) {\n            this.setAttribute('trigger', value);\n        }\n        else {\n            this.removeAttribute('trigger');\n        }\n    }\n    get trigger() {\n        return this.getAttribute('trigger');\n    }\n    set speed(value) {\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(value)) {\n            this.removeAttribute('speed');\n        }\n        else {\n            this.setAttribute('speed', value);\n        }\n    }\n    get speed() {\n        if (this.hasAttribute('speed')) {\n            return parseFloat(this.getAttribute('speed'));\n        }\n        return null;\n    }\n    set stroke(value) {\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(value)) {\n            this.removeAttribute('stroke');\n        }\n        else {\n            this.setAttribute('stroke', value);\n        }\n    }\n    get stroke() {\n        if (this.hasAttribute('stroke')) {\n            return parseFloat(this.getAttribute('stroke'));\n        }\n        return null;\n    }\n    set scale(value) {\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(value)) {\n            this.removeAttribute('scale');\n        }\n        else {\n            this.setAttribute('scale', value);\n        }\n    }\n    get scale() {\n        if (this.hasAttribute('scale')) {\n            return parseFloat(this.getAttribute('scale'));\n        }\n        return null;\n    }\n    set axisX(value) {\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(value)) {\n            this.removeAttribute('axis-x');\n        }\n        else {\n            this.setAttribute('axis-x', value);\n        }\n    }\n    get axisX() {\n        if (this.hasAttribute('axis-x')) {\n            return parseFloat(this.getAttribute('axis-x'));\n        }\n        return null;\n    }\n    set axisY(value) {\n        if ((0,_helpers_utils_js__WEBPACK_IMPORTED_MODULE_2__.isNil)(value)) {\n            this.removeAttribute('axis-y');\n        }\n        else {\n            this.setAttribute('axis-y', value);\n        }\n    }\n    get axisY() {\n        if (this.hasAttribute('axis-y')) {\n            return parseFloat(this.getAttribute('axis-y'));\n        }\n        return null;\n    }\n    /**\n     * Access animation container element.\n     */\n    get container() {\n        return this._root.lastElementChild;\n    }\n    /**\n     * Access icon data for this element.\n     */\n    get iconData() {\n        if (this.icon && typeof this.icon === \"object\") {\n            return this.icon;\n        }\n        return (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.getIcon)(this.icon || this.src);\n    }\n    /**\n     * Current animation speed.\n     */\n    get animationSpeed() {\n        if (this.hasAttribute('speed')) {\n            const v = this.getAttribute('speed');\n            return v === null ? 1 : parseFloat(v);\n        }\n        return 1;\n    }\n    /**\n     * Check element version.\n     */\n    get version() {\n        return _global_js__WEBPACK_IMPORTED_MODULE_1__.VERSION;\n    }\n}\n//# sourceMappingURL=element.js.map\n\n//# sourceURL=webpack://lord-icon-bundle/./node_modules/lord-icon-element/build/main/element.js?");
```

Seems possibly related to `__webpack_require__`:
```
__webpack_require__(/*! ./main/element.js */ "./node_modules/lord-icon-element/build/main/element.js");
```

**Comment by jjallaire** [comment]
you are going to need to build a non-minified version in order to see what the problem is

**Comment by jmgirard** [comment]
I had thought that bundling on development mode was non-minified. Is that not the case?

**Comment by jmgirard** [comment]
> Uncaught SyntaxError: Unexpected token ;

Error appears to be on line 965 of non-minified output (the line reading `_root;`):

    class Element extends HTMLElement {
        _root;
        _isReady = false;
        _lottie;
        _properties;
        _connectedTrigger;
        _storedIconData;
        _palette;
        /**
         * Register Lottie library.
         * @param animationLoader Provide "loadAnimation" here from Lottie.
         */
        static registerAnimationLoader(animationLoader) {
            (0,_manager_js__WEBPACK_IMPORTED_MODULE_0__.registerAnimationLoader)(animationLoader);
        }

**Comment by jjallaire** [comment]
That's some ES class syntax that is likely not supported in RStudio. You need to coax Babel into emitting something different (not sure how exactly to do this though!)

**Comment by jmgirard** [comment]
Ok, I'll keep looking. Thanks! (By the way, do you know if RStudio plans to support ES6+ in the future?)

**Comment by jjallaire** [comment]
Yes, the current daily build (to be release in November) does indeed
support ES6+

**Comment by cscheid** [comment]
Just in case: other ES syntax features you'll need to disable include the nullish coalescing operator `??` and the optional chaining operator `?.`. Those two tend to show up on NPM packages and RStudio does not currently support them.

**Comment by jmgirard** [comment]
Thanks for the heads up. I'm assuming that there must be a way to tell babel to disable specific features. Looking into that later today.

**Comment by jmgirard** [comment]
Ok, I can confirm that the issue is resolved when using the latest RStudio daily build.

> RStudio 2022.10.0-daily+69 "Elsbeth Geranium" Daily (e07e7bcd5133850c4cff145c343cfe7103e642dc, 2022-07-15) for Windows
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2022.10.0-daily+69 Chrome/102.0.5005.148 Electron/19.0.8 Safari/537.36

I didn't have much luck with babel. It seems that the current presets tend to go to ES2015/ES6+ but the current stable version of RStudio seems to want ES5.

**Comment by jmgirard** [comment]
This is now fixed in RStudio version 2022.12.0+353.
